lib/spack/spack/fetch_strategy.py

```diff
@@ -267,7 +267,10 @@ def __init__(self, url=None, checksum=None, **kwargs):
     @property
     def curl(self):
         if not self._curl:
-            self._curl = which('curl', required=True)
+            try:
+                self._curl = which('curl', required=True)
+            except CommandNotFoundError as exc:
+                tty.error(str(exc))
```

Comment:
Just to make sure I understand what's going on: This is because the calling code is a try/except block that catches the error?

Reply:
Basically yes.  This is wrapped in `try-except` to ensure the error is output since `stage.fetch` only prints what is considered a significant issue in `debug` mode.  `URLFetchStrategy`'s `fetch` will fail in `_existing_url` when `self._curl` is not set so the `candidate_urls` loop terminates early with the error in the test code.